internal/introspection/idlmodule.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package introspection
+
+import "strings"
+
+// IDLModule is a generic IDL module. For example, a thrift file or a protobuf
+// one.
+type IDLModule struct {
+	FilePath   string      `json:"filePath"`
+	SHA1       string      `json:"sha1"`
+	Includes   []IDLModule `json:"includes,omitempty"`
+	RawContent string      `json:"rawContent,omitempty"`
+}
+
+// Flatmap iterates recursively over the idl module and its includes. The cb
+// function is called onces for every unique idl module (based on FilePath).
+// Iteration is aborted if cb returns true.
+func (im *IDLModule) Flatmap(cb func(*IDLModule) bool) {
+	ft := newIDLFlattener(cb)
+	ft.Collect(im)
+}
+
+// IDLModules is a sortable slice of IDLModule.
+type IDLModules []IDLModule
+
+func (ims IDLModules) Len() int {
```

Comment:
I won't comment on every one of these, but as I said, in general you need a godoc comment on every public type.